internal/jobs/queue.go

```diff
@@ -49,8 +50,37 @@ func Initialize(ctx context.Context, logger *zerolog.Logger) error {
 	return nil
 }
 
+func Enqueue(ctx context.Context, jobs ...dejq.PendingJob) error {
+	if queue == nil {
+		panic("job queue was not initialized yet, for tests import internal/jobs/stub")
+	}
+	return queue.Enqueue(ctx, jobs...)
+}
+
 func StartDequeueLoop(ctx context.Context, logger *zerolog.Logger) {
 	logger.Debug().Msg("Starting dequeue loop")
 	ctx = ctxval.WithLogger(ctx, logger)
-	Queue.DequeueLoop(ctx)
+	queue.DequeueLoop(ctx)
+}
+
+func StopDequeueLoop() {
+	queue.Stop()
+}
+
+func HandleStub(_ context.Context, _ dejq.Job) error {
+	return nil
+}
+
+func InitializeStub() {
+	var err error
+	ctx := context.Background()
+	queue, err = mem.NewClient(ctx, zerologr.New(&log.Logger))
+	if err != nil {
+		panic(err)
+	}
+	queue.RegisterHandler(TypeNoop, HandleStub)
+	queue.RegisterHandler(TypePubkeyUploadAws, HandleStub)
+	queue.RegisterHandler(TypeLaunchInstanceAws, HandleStub)
+	queue.DequeueLoop(ctx)
+	log.Logger.Debug().Msg("initialized job queue stub")
```

Comment:
Unfortunately, we need to keep this list of handlers up to date with the production list above. I can perhaps implement a flag in the dejq memory implementation to silently ignore when a handler is not registered, that would do it. Opinions?

Reply:
I believe it should not be as bad to keep the list, we have the same for dao, if you wanna test something that needs enque something, you need to stub it's dependencies, one line dependency is ok IMHO :shrug:

Reply:
Actually, this was not valid comment. I figured out (see above) that the memory implementation silently skips unregistered handlers so I am going to remove these registrations and `HandleStub` completely! https://github.com/lzap/dejq/blob/master/mem/mem_client.go#L75